### PR TITLE
Minor refactoring of scheduler

### PR DESCRIPTION
**Pull request by naoyam**
Taken from #1772. Originally done by @csarofeen.

Nothing should change with this PR. Just cleaning up.

- [x] Benchmark performance confirmation

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/scheduler/registry.cpp [hunk 1]
```diff
@@ -1807,54 +1807,72 @@ class PersistentKernelScheduler : public SchedulerEntry {
...
     // If there's a small iteration dimension but a large reduction dimension it
     // may not make sense to make a persistent kernel
     auto properties =
         scheduler_utils::getProperties(fusion, runtime_info, reduction_tvs[0]);
 
+    // TODO: Enable grid persistence
+    const auto available_persistent_buffer_size =
+        scheduler_utils::register_file_size;
+
+    if (persistent_buffer_size > available_persistent_buffer_size) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/scheduler/registry.cpp:1822]: This should prevent the scheduler from getting a grid persistent kernel for now, as we need to multiple the persistent buffer by the SM count, right?

**Review by csarofeen — APPROVED** [review]
Seems fine to me as it won't change anything today. Not sure how good my logic was in registry, but I assume you'll work on that as part of heuristics/scheduling.

File: torch/csrc/jit/codegen/cuda/scheduler/registry.cpp (hunk 1 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/scheduler/registry.cpp:1822]: Yes, I just took part of the registry changes that's not specific to outer grid reductions.

**Comment by naoyam** [comment]
@csarofeen I noticed some slight changes in scheduling results because of this:

```
// SMs that can be used simultaneously
        properties.total_iteration_numel * required_sm_per_norm <
            min_fraction_of_sms)
```

The corresponding part in `devel` is:

```
properties.total_iteration_numel <
                (properties.fastest_dim_reduction
                     ? std::max(device_multiprocessor_count / 8, (int64_t)1)
                     // Make sure we at least use a quarter of the device * a
                     // half warp
                     : (warp_size / 8) * device_multiprocessor_count) &&
```

`min_fraction_of_sms` is `safeDiv(device_multiprocessor_count, 8)`. For block persistence, `required_sm_per_norm` is always 1. So, the new condition is just saying the iteration count is smaller than 1/8 of the SM count. In the current devel case, the right-hand side of the comparison is multiplied by `(warp_size / 8)`, so it's 4 times the SM count. The persistent scheduler rejects when these conditions are met, so the new one uses the persistent scheduler much more smaller problems.

I'm not sure what the optimal heuristics would be, but I don't want to change any heuristics decision in this PR, so I reverted the change ([d05187b](https://github.com/csarofeen/pytorch/pull/2253/commits/d05187b61addeb82881db92db02763371fb7e6ce))

**Pull request merged**